aten/src/ATen/record_function.cpp

```diff
@@ -428,4 +457,41 @@ void RecordFunction::end() {
   active_ = false;
 }
 
+// RecordFunction pre-sampling
+namespace {
+// Whether to try to create RecordFunction on each call (>0) or
+// use pre-sampling (=0)
+std::atomic<int> global_record_all_functions_ {0};
+}
+
+void setRecordAllFunctions() {
+  ++global_record_all_functions_;
+}
+void unsetRecordAllFunctions() {
+  TORCH_CHECK(--global_record_all_functions_ >= 0);
+}
+
+bool shouldRunRecordFunction(bool& pre_sampled) {
+  auto* rf_tls_ptr = &rf_tls_;
+  if (!rf_tls_ptr->tls_record_function_enabled_) {
+    pre_sampled = false;
+    return false;
```

Comment:
If you return false here, is there any point in setting `pre_sampled`?

Reply:
then we should say 'if function returns false, pre_sampled argument value is unchanged/uninitialized', i slightly prefer to always have a deterministic out value